app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/LinkPreviewViewHolder.java

```diff
@@ -378,4 +330,62 @@ public void onFailure(Call<SkillRatingResponse> call, Throwable t) {
 
         });
     }
+
+    private ResponseListener setResponseListener() {
+
+        return responseListener = new ResponseListener() {
+            @Override
+            public void onData(MetaData data) {
+                if (!PrefManager.hasTokenExpired() || PrefManager.getBoolean(Constant.ANONYMOUS_LOGGED_IN, false)) {
+                    realm.beginTransaction();
+                    Realm realm = Realm.getDefaultInstance();
```

Comment:
You began a transaction and then got a new instance of Realm?

Reply:
@iamareebjamal this was the previous code that was already present I just shifted it to a new function, I will work upon it now

Reply:
No, let it be. Just handle the changes above